catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/app/WhenConditionBrickTest.java

```diff
@@ -0,0 +1,91 @@
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.uiespresso.content.brick.app;
+
+import android.support.test.InstrumentationRegistry;
+import android.support.test.runner.AndroidJUnit4;
+
+import org.catrobat.catroid.ProjectManager;
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.content.Project;
+import org.catrobat.catroid.content.Script;
+import org.catrobat.catroid.content.Sprite;
+import org.catrobat.catroid.content.WhenConditionScript;
+import org.catrobat.catroid.content.bricks.NoteBrick;
+import org.catrobat.catroid.formulaeditor.Formula;
+import org.catrobat.catroid.formulaeditor.FormulaElement;
+import org.catrobat.catroid.formulaeditor.Operators;
+import org.catrobat.catroid.ui.SpriteActivity;
+import org.catrobat.catroid.uiespresso.testsuites.Cat;
+import org.catrobat.catroid.uiespresso.testsuites.Level;
+import org.catrobat.catroid.uiespresso.util.rules.BaseActivityInstrumentationRule;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+import static org.catrobat.catroid.uiespresso.content.brick.utils.BrickDataInteractionWrapper.onBrickAtPosition;
+
+@RunWith(AndroidJUnit4.class)
+public class WhenConditionBrickTest {
+	@Rule
+	public BaseActivityInstrumentationRule<SpriteActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(SpriteActivity.class, SpriteActivity.EXTRA_FRAGMENT_POSITION, SpriteActivity.FRAGMENT_SCRIPTS);
+
+	@Before
+	public void setUp() throws Exception {
+		createProject();
+		baseActivityTestRule.launchActivity();
+	}
+
+	@Category({Cat.CatrobatLanguage.class, Level.Smoke.class})
+	@Test
+	public void whenConditionBrick() {
+		onBrickAtPosition(0).checkShowsText(R.string.brick_when_condition_when);
+		onBrickAtPosition(1).checkShowsText(R.string.brick_note);
+
+		onBrickAtPosition(0).onFormulaTextField(R.id.brick_when_condition_edit_text)
+				.check(matches(withText("1 < 2 ")));
```

Comment:
the space at the end is a longer story -> it has to do with the way formulas are parsed (it's the same for all other formula fields and it has nothing to do with the refactoring).

this test will fail whenever the issue described in CATROID-52 reappears.

Reply:
yes, that was part of the bug (that it was a string). 

no, you cannot, because the test was added in this commit.

Reply:
I know that it was part of the bug, but in my case, the string had no space at the end. Why is there this difference? I am talking about 0.9.41 "Code Nightly #3". No space, just '1 < 2' as a string in the field, incorrectly.

As for the test: I must trust you that you checked it. Can I rely on this? Do you rely on that when you make code reviews for PRs by other devs, i.e., that they really run the tests before the correction, and that the failed as expected and as it should be?

Reply:
before the fix the string was displayed as followed:
"'1 < 2' "
due to the workings of our formula, there is a space at the end of the representation of a formula, no matter if it is a number, string, or whole formula. formula 1 would be displayed as "1 ", formula "Some string" would be displayed "Some string ".

the test would fail because the EditText would contain "'1 < 2' " instead of "1 < 2 ".

i wrote the test, while thmq was writing the fix, i verified the test is indeed failing.

regarding code reviews in general:
yes, we do rely on developers doing proper TDD, but the red-green-refacotor workflow happens on developers machine and does NOT mix with our review workflow.

Reply:
OK, thanks, that answers all my questions. Sorry for the confusion and angst on my part ;-) 
Please now move the ticket to PO review, we will mark it as merged.